### PR TITLE
Implement new Button Examples

### LINKED ISSUE #89: Implement new Button Screen
Contributes to #14 

## Use cases
 - [x] Button with default styling
   - Submit Application
 - [x] Button with color="red"
   - Cancel Application
- [x] Buttons with container style as justifyContent: 'space-between'
   - Cancel (Can be Red in colour)
   - Save for later (Green Colour)
- [x] Buttons with container style as justifyContent: 'space-evenly'
   - Cancel (Red colour)
   - Save for later (Blue)
   - Submit (Green Colour)
- [x] Button with disabled={true}
   - Submit Application
- [x] Button accessibilityLabel="label" (No visual differences)
   - Submit Application
- [x] Button with no onPress
   - Submit Application

## Doubts
 - Other props like nextFocusDown, touchSoundDisabled can be added but no visual differences and testing strategy. 

## E2E Tests
 - [x] Rendering
 - [x] Simple button should be tappable
 - [x] Adjusted color button should be tappable
 - [x] Button with accessibilityLabel should be tappable
 - [x] Two buttons with JustifyContent:'space-between' should be tappable
 - [x] Three buttons with JustifyContent:'space-evenly' should be tappable
 - [x] Disabled button should not interact
 - [x] Button with no onPress should not interact 

## suggestion
 - Use cases should be like 'real' app-like cases
 - Refer to gist sent on discord

### PR DESCRIPTION
**Pull request by chirag-singhal**
Fixes #89 

## Changelog
 * Added new use cases for button component
 * Added detox tests for these new use cases

### CODE REVIEW

File: packages/RNTester/.detoxrc.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
       "device": {
-        "avdName": "testAVD"
+        "avdName": "Pixel_API_28"
       }
     },
     "android.emu.release": {
       "binaryPath": "android/app/build/outputs/apk/release/app-release.apk",
-      "build": "cd android && ./gradlew assembleRelease assembleAndroidTest -DtestBuildType=release && cd ..",
+      "build": "cd android && ./gradlew assembleRelease app:assembleAndroidTest -DtestBuildType=release && cd ..",
       "type": "android.emulator",
```
Comment (YashKumarVerma) [inline packages/RNTester/.detoxrc.json:16]: these changes have already been merged, will it create any issues while merging?

File: packages/RNTester/js/examples/Button/ButtonExample.js [hunk 2]
```diff
@@ -26,92 +25,133 @@ exports.description = 'Simple React Native button component.';
...
-                accessibilityLabel="See an informative alert"
-              />
-            );
-          }}
-        </RNTesterThemeContext.Consumer>
+        <Button
+          onPress={() => onButtonPress('submitted')}
+          testID="button_default_styling"
+          title="Submit Application"
+          accessibilityLabel="Press to submit your application!"
```
Comment (YashKumarVerma) [inline packages/RNTester/js/examples/Button/ButtonExample.js:37]: The remaining files have a single quote styling in strings, could we have it here also?

Comment (chirag-singhal) [reply packages/RNTester/js/examples/Button/ButtonExample.js:36]: @YashKumarVerma done

File: packages/RNTester/.detoxrc.json (hunk 1 above)
Comment (chirag-singhal) [reply packages/RNTester/.detoxrc.json:16]: @YashKumarVerma i don't think so they should

File: packages/RNTester/metro.config.js [hunk 3]
```diff
@@ -6,7 +6,7 @@
  */
 
 const path = require('path');
-const reactNativePath = path.resolve('../..');
+const reactNativePath = path.resolve('./node_modules/react-native');
```
Comment (anku255) [inline packages/RNTester/metro.config.js:9]: Hey @chirag-singhal! This is just a hack to make RNTester/Detox tests work on our machine locally. This should not be committed.

Comment (chirag-singhal) [reply packages/RNTester/metro.config.js]: @anku255 Done :+1:

File: packages/RNTester/js/examples/Button/ButtonExample.js [hunk 4]
```diff
@@ -26,92 +25,131 @@ exports.description = 'Simple React Native button component.';
 
 exports.examples = [
   {
-    title: 'Simple Button',
-    description: ('The title and onPress handler are required. It is ' +
-      'recommended to set accessibilityLabel to help make your app usable by ' +
-      'everyone.': string),
+    title: 'Button with default styling',
+    description: '',
```
Comment (AnshG714) [inline packages/RNTester/js/examples/Button/ButtonExample.js:29]: You don't need to provide an empty string to the description, it's an optional field.

**Pull request merged**